healpy/test/test_fitsfunc.py

```diff
@@ -311,5 +311,32 @@ def test_getformat():
     assert getformat(["DD", "CCC"]) == "A3"
 
 
+def test_writemap_newdefault(tmp_path):
+    filename = tmp_path / "test_map.fits"
+    for dtype in (
+        "bool",
+        "uint8",
+        "int16",
+        "int32",
+        "int64",
+        "float32",
+        "float64",
+        "complex64",
+        "complex128",
+    ):
+        m = np.ones(12, dtype=dtype)
+        write_map(filename, m, overwrite=True)
+        with pf.open(filename) as f:
+            assert f[1].data.field(0).dtype.name == dtype
+            np.testing.assert_array_almost_equal(f[1].data.field(0), m)
+
+
+def test_writemap_typestr(tmp_path):
+    filename = tmp_path / "test_map.fits"
+    m = np.zeros(12, dtype=">f4")
+    write_map(filename, m, overwrite=True)
+    assert pf.open(filename)[1].data.field(0).dtype.name == "float32"
```

Comment:
This doesn't test that endianness is handled correctly. To do that, you need to write an asymmetric byte sequence: `np.ones(12)` would do the trick. Then you need to check that the bytes of the file written to disk are identical regardless of the endianness of the Numpy array.

Reply:
ok, done, it is more important that what I read back is the correct array, so I am testing that.